### PR TITLE
[controller] fix topic priority determination in TopicCleanupService

### PR DESCRIPTION
**Pull request by arjun4084346**
## Summary, imperative, start upper case, don't end with a period

The comparator used in TopicCleanupService to assign higher priority to RT topics over non RT topics `(s1, s2) -> s1.isRealTime() ? -1 : 0` is incorrect. It does not assign two RT topics (or two non RT topics) as equal priority, which violates the fundamental contract of sorting algorithms.
In some cases, this results in RT topic being assigned lower priority than a non RT topic. This PR fixes the comparator and uses both the topics in determining the relative priority.

Resolves #XXX

## How was this PR tested?

added a unit test in PubSubTopicPartitionInfoTest, which fails without the fix.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestHybridStoreDeletion.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ public void testHybridStoreRTDeletionWhileIngesting() {
 
       createStoresAndVersions(storeNames, streamingRewindSeconds, streamingMessageLag);
 
-      // Wait until the rt topic of the first store is fully deleted.
+      // Wait until the rt topic of the first store is fully created.
```
Comment (arjun4084346) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestHybridStoreDeletion.java:127]: this file's changes are unrelated to the fix.

File: internal/venice-common/src/test/java/com/linkedin/venice/pubsub/PubSubTopicPartitionInfoTest.java [hunk 2]
```diff
@@ -36,4 +40,31 @@ public void testToString() {
...
+    PubSubTopic topic3 = new PubSubTopicImpl("topic3" + Version.REAL_TIME_TOPIC_SUFFIX);
+    PubSubTopic topic4 = new PubSubTopicImpl("topic4");
+    PubSubTopic topic5 = new PubSubTopicImpl("topic5");
+    PubSubTopic topic6 = new PubSubTopicImpl("topic6" + Version.REAL_TIME_TOPIC_SUFFIX);
+    allTopics.add(topic1);
+    allTopics.add(topic2);
+    allTopics.add(topic3);
+    allTopics.add(topic4);
+    allTopics.add(topic5);
+    allTopics.add(topic6);
```
Comment (lluwm) [inline internal/venice-common/src/test/java/com/linkedin/venice/pubsub/PubSubTopicPartitionInfoTest.java:59]: Is it possible to add some `randomness` into this testing? For example, let's say, in a for loop, we generate real time and version topics based on probability, and count their appearances respectively. In the verification phase, we poll and assert them.

**Review by lluwm — APPROVED** [review]
LGTM! Thanks @arjun4084346 (congrats on your first Venice PR)

**Review by KaiSernLim — APPROVED** [review]
